audio-worklet/src/noisegate-script-processor.js

```diff
@@ -0,0 +1,221 @@
+    this.input.connect(this.noiseGateKernel_).connect(this.output);
+
+    // The previous envelope level, a float representing signal amplitude.
+    this.previousLevel_ = 0;
+
+    // The last weight (between 0 and 1) assigned, where 1 means the gate
+    // is open and 0 means it is closed and the sample in the output buffer is
+    // muted. When attacking, the weight will linearly decrease from 1 to 0, and
+    // when releasing the weight linearly increase from 0 to 1.
+    this.previousWeight_ = 1.0;
+    this.channel_ = new Float32Array(this.noiseGateKernel_.bufferSize);
+    this.envelope_ = new Float32Array(this.noiseGateKernel_.bufferSize);
+    this.weights_ = new Float32Array(this.noiseGateKernel_.bufferSize);
+  }
+
+  /**
+   * Control the dynamic range of input based on specified threshold.
+   * @param {AudioProcessingEvent} event An Event object containing
+   *                                     input and output buffers
+   */
+  onaudioprocess_(event) {
+    let inputBuffer = event.inputBuffer;
+    let channel0 = inputBuffer.getChannelData(0);
+
+    // Stereo input is downmixed to mono input via averaging.
+    if (inputBuffer.numberOfChannels === 2) {
+      let channel1 = inputBuffer.getChannelData(1);
+      for (let i = 0; i < channel1.length; i++) {
+        this.channel_[i] = (channel0[i] + channel1[i]) / 2;
+      }
+    } else {
+      this.channel_ = channel0;
+    }
+
+    let envelope = this.detectLevel_(this.channel_);
+    let weights = this.computeWeights_(envelope);
+    
+    for (let i = 0; i < inputBuffer.numberOfChannels; i++) {
+      let input = inputBuffer.getChannelData(i);
+      let output = event.outputBuffer.getChannelData(i);
+
+      for (let j = 0; j < input.length; j++) {
+        output[j] = weights[j] * input[j];
+      }
+    }
+  }
+
+  /**
+   * Detect level using the difference equation for the Root Mean Squared
+   * value of the input signal.
+   * @param {Float32Array} channelData Input channel data.
+   * @return {Float32Array} The level of the signal, the signal's amplitude.
+   */
+  detectLevel_(channelData) {
+    // The signal level is determined by filtering high frequency oscillation
+    // with exponential smoothing.
+    // This is equivalent to computing the root-mean-square (RMS) value
+    // of the signal. See http://www.aes.org/e-lib/browse.cfm?elib=16354 for
+    // details.
+    this.envelope_[0] = this.alpha_ * this.previousLevel_ +
+        (1 - this.alpha_) * Math.pow(channelData[0], 2);
+
+    for (let j = 1; j < channelData.length; j++) {
+      this.envelope_[j] = this.alpha_ * this.envelope_[j - 1] +
+          (1 - this.alpha_) * Math.pow(channelData[j], 2);
+    }
+    this.previousLevel_ = this.envelope_[this.envelope_.length - 1];
+    
+    return this.envelope_;
+  }
+
+  /**
+   * Computes an array of weights which determines what samples are silenced.
+   * @param {Float32Array} envelope Array of amplitudes from envelope follower.
```

Comment:
This is not an array of amplitudes anymore.  Maybe just say this is the output of the envelope follower.